invokeai/app/invocations/latent.py

```diff
@@ -310,17 +318,37 @@ def prep_control_data(
                 # MultiControlNetModel has been refactored out, just need list[ControlNetData]
         return control_data
 
+    def init_scheduler(self, scheduler, device, steps, denoising_start, denoising_end):
+        # apply denoising_start
+        num_inference_steps = steps
+        scheduler.set_timesteps(num_inference_steps, device=device)
+
+        t_start = int(round(denoising_start * num_inference_steps))
+        timesteps = scheduler.timesteps[t_start * scheduler.order :]
+        num_inference_steps = num_inference_steps - t_start                
+
+        # apply denoising_end
+        num_warmup_steps = max(len(timesteps) - num_inference_steps * scheduler.order, 0)
+
+        skipped_final_steps = int(round((1 - denoising_end) * steps))
+        num_inference_steps = num_inference_steps - skipped_final_steps
+        timesteps = timesteps[: num_warmup_steps + scheduler.order * num_inference_steps]
+
+        return num_inference_steps, timesteps
+
+
     @torch.no_grad()
     def invoke(self, context: InvocationContext) -> LatentsOutput:
         with SilenceWarnings():
             noise = context.services.latents.get(self.noise.latents_name)
+            seed = self.noise.seed or 0
```

Comment:
This works but is a bit unintuitive - if `seed == 0`, we end up using the right-hand side to set it to `0` again.

Prefer checking for `None` and assigning to zero, even though functionally it's identical

Reply:
Ok... I thought it's not uncommon patter, but I'll change to if